bindings/c/lib.cpp

```diff
@@ -185,14 +186,8 @@ kore_pattern *kore_pattern_new_token_with_len(
 
 kore_pattern *kore_pattern_new_injection(
     kore_pattern const *term, kore_sort const *from, kore_sort const *to) {
-  auto inj_sym = kore_symbol_new("inj");
-  kore_symbol_add_formal_argument(inj_sym, from);
-  kore_symbol_add_formal_argument(inj_sym, to);
-
-  auto inj = kore_composite_pattern_from_symbol(inj_sym);
-  kore_composite_pattern_add_argument(inj, term);
-
-  kore_symbol_free(inj_sym);
+  auto inj = new kore_pattern;
+  inj->ptr_ = kllvm::bindings::make_injection(term->ptr_, from->ptr_, to->ptr_);
```

Comment:
@dwightguth As an example of how to look at the changes in this diff, the `kllvm::bindings::make_injection` function contains code that is _equivalent_ to the C on the LHS here, but using the underlying C++ structures directly. All the changes in this commit are of that form.